src/PageContent/Directory/DirectoryPage.js

```diff
@@ -0,0 +1,256 @@
+const logoURL = 'https://upload.wikimedia.org/wikipedia/commons/8/89/Portrait_Placeholder.png';
+
+class DirectoryPage extends Component {
+    
+    constructor(props) {
+        super(props);
+        this.state = {
+            umbrellaId: '',
+            raList: [],
+            dgList: [],
+            daList: [],
+        };
+    }
+
+    async componentDidMount() {
+        let umbrellaId = await this.getUmbrellaId();
+        await this.setState({umbrellaId: umbrellaId});
+        this.fetchOrgs();
+    }
+
+    componentWillUnmount() {
+        // detach listeners
+        db.ref('accounts').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+        db.ref('donating_agencies').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+    }
+
+    async getUmbrellaId() {
+        const { account }= this.props;
+        switch (account.accountType) {
+        case AccountType.DONATING_AGENCY_MEMBER: { // get umbrellaId from db.ref(`donating_agencies)
+            let daSnapshot = await db.ref(`donating_agencies/${account.agency}`).once('value');
+            let { umbrella } = daSnapshot.val();
+            return umbrella;
+        }
+        case AccountType.RECEIVING_AGENCY:
+            return account.umbrella;
+        case AccountType.DELIVERER_GROUP:
+            return account.umbrella;
+        case AccountType.UMBRELLA: // if the currently logged in account is umbrella, return userId of this account
+            return account.uid;
+        default:
+            return;
+        }
+    }
+
+    fetchOrgs() {
+        switch (this.props.account.accountType) {
+        case AccountType.DONATING_AGENCY_MEMBER:
+            this.fetchRaAndDg([AccountType.DELIVERER_GROUP, AccountType.RECEIVING_AGENCY]); 
+            break;
+        case AccountType.RECEIVING_AGENCY:
+            this.fetchRaAndDg([AccountType.DELIVERER_GROUP]); 
+            this.fetchDAM();
+            break;
+        case AccountType.DELIVERER_GROUP:
+            this.fetchRaAndDg([AccountType.RECEIVING_AGENCY]);
+            this.fetchDAM();
+            break;
+        case AccountType.UMBRELLA:
+            this.fetchRaAndDg([AccountType.RECEIVING_AGENCY, AccountType.DELIVERER_GROUP]);   
+            this.fetchDAM();
+            break;
+        default:
+            break;
+        }
+    }
+
+    fetchRaAndDg(orgTypes) { // fetch receiving agencies or delivery groups
+        // keeps listening for real-time database change 
+        db.ref('accounts').orderByChild('umbrella').equalTo(this.state.umbrellaId).on('value', (snapshot) => { 
+            let accountObjects = snapshot.val();       
+            let raList = [];
+            let dgList = [];
+
+            // loop through all AccountObjects
+            for (let key in accountObjects) {
+                let accountItem = accountObjects[key];
+                if (accountObjects.hasOwnProperty(key)  // filter out prototype props
```

Comment:
what does this do?

Reply:
It loops through each object in `accountObjects`. `accountObjects.hasOwnProperty(key` is to filter out `prototype props` (if there's any) of an javascript object.